test/integration/kubernetes_deploy_test.rb

```diff
@@ -686,4 +686,19 @@ def test_ejson_secrets_respects_no_prune_flag
     ejson_cloud.assert_secret_present('catphotoscom', type: 'kubernetes.io/tls', managed: true)
     ejson_cloud.assert_secret_present('ejson-keys', managed: false)
   end
+
+  def test_invalid_context
+    old_config = ENV['KUBECONFIG']
+    begin
+      ENV['KUBECONFIG'] = File.join(__dir__, '../fixtures/invalid_config.yml')
+      result = deploy_fixtures('hello-cloud')
+      assert_deploy_failure(result)
+      assert_logs_match_all([
```

Comment:
These assertions are a good start, but I'd like to make them more strict re: what we're printing in what order. Eg:
```
[
  "The following command failed: kubectl version",
  "Unable to connect to the server",
  "Result: FAILURE",
  "Failed to reach server for minikube"
], in_order: true
```

And maybe an extra assertion specifically on "Unable to connect to the server" being logged X times to demonstrate that we not only retried but also showed the message each time to help with debugging.